src/Lecture3.hs

```diff
@@ -82,7 +87,11 @@ Tuesday
   would work for **any** enumeration type in Haskell (e.g. 'Bool',
   'Ordering') and not just 'Weekday'?
 -}
-next = error "TODO"
+next :: (Enum a) => a -> a
+next a = case enumFrom a of
+  [] -> toEnum (0 :: Int)
+  [_] -> toEnum (0 :: Int)
+  (_:s:_) -> s
```

Comment:
That's an interesting implementation using `enumFrom` 🙂 
Alternatively, you can use `succ` and `maxBound / minBound` to avoid allocating memory for lists